backend/app/api/admin_routes/evaluation/evaluation_dataset.py

```diff
@@ -0,0 +1,233 @@
+def update_evaluation_dataset(
+    evaluation_dataset_id: int,
+    updated_evaluation_dataset: UpdateEvaluationDataset,
+    session: SessionDep,
+    user: CurrentSuperuserDep
+) -> EvaluationDataset:
+    evaluation_dataset = session.get(EvaluationDataset, evaluation_dataset_id)
+
+    if not evaluation_dataset or evaluation_dataset.user_id != user.id:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Evaluation dataset not found",
+        )
+
+    evaluation_dataset.name = updated_evaluation_dataset.name
+
+    session.merge(evaluation_dataset)
+    session.commit()
+
+    return evaluation_dataset
+
+
+@router.get("/admin/evaluation/dataset")
+def list_evaluation_dataset(
+    session: SessionDep,
+    user: CurrentSuperuserDep,
+    params: Params = Depends(),
+) -> Page[EvaluationDataset]:
+    stmt = (
+        select(EvaluationDataset)
+        .where(EvaluationDataset.user_id == user.id)
+        .order_by(desc(EvaluationDataset.id))
+    )
+    return paginate(session, stmt, params)
+
+
+@router.post("/admin/evaluation/dataset-item")
+def create_evaluation_dataset_item(
+    evaluation_dataset_item: ModifyEvaluationDatasetItem,
+    session: SessionDep,
+    user: CurrentSuperuserDep
+) -> EvaluationDatasetItem:
+    evaluation_task_item = EvaluationDatasetItem(
+        query=evaluation_dataset_item.query,
+        reference=evaluation_dataset_item.reference,
+        retrieved_contexts=evaluation_dataset_item.retrieved_contexts,
+        extra=evaluation_dataset_item.extra,
+        evaluation_dataset_id=evaluation_dataset_item.evaluation_dataset_id,
+    )
+
+    session.add(evaluation_task_item)
+    session.commit()
+
+    return evaluation_task_item
+
+
+@router.delete("/admin/evaluation/dataset-item/{evaluation_dataset_item_id}")
+def delete_evaluation_dataset_item(
+        evaluation_dataset_item_id: int,
+        session: SessionDep,
+        user: CurrentSuperuserDep
+) -> bool:
+    evaluation_dataset_item = session.get(EvaluationDatasetItem, evaluation_dataset_item_id)
+
+    if not evaluation_dataset_item:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Evaluation dataset item not found",
+        )
+
+    session.delete(evaluation_dataset_item)
+    session.commit()
+
+    return True
+
+
+@router.put("/admin/evaluation/dataset-item/{evaluation_dataset_item_id}")
+def update_evaluation_dataset_item(
+    evaluation_dataset_item_id: int,
+    updated_evaluation_dataset_item: ModifyEvaluationDatasetItem,
+    session: SessionDep,
+    user: CurrentSuperuserDep
+) -> EvaluationDatasetItem:
+    evaluation_dataset_item = session.get(EvaluationDatasetItem, evaluation_dataset_item_id)
+
+    if not evaluation_dataset_item:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Evaluation dataset item not found",
+        )
```

Comment:
Can we wrap this duplicate code in the `evaluation_dataset_repo.must_get`? like: 

https://github.com/pingcap/autoflow/blob/98b8686497f567773bd9f874966f6e7a2aff41f3/backend/app/repositories/embedding_model.py#L47

Reply:
It doesn't have a repository for evaluation tables, use a generic function instead.

Reply:
Why not create one?

Reply:
Pluralitas non est ponenda sine necessitate.